UseCases.md

```diff
@@ -0,0 +1,100 @@
+This file describes exact behavior of methods for different edge cases and 
+explains general logic. This description covers the behavior of get_tld, 
+get_tld_unsafe, get_sld, get_sld_unsafe, split_domain, split_domain_unsafe
+
+1. general difference of get_*() and get_*_unsafe() methods:
+get_*_unsafe() does not perform if the input string is None and does not 
+transforms it to the lower case.
+
+2. The listed above methods works only with non-canonical FQDN strings - 
+trailing dot must be removed before call the method. This restriction allows 
+get rid of fuzzy logic in edge cases.
+
+3. DNS does not support empty labels - if some label detected to be empty, 
+None will be returned. 
+
+4. Every method processes provided FQDN in the reverse order, from the last 
+label towards the start of the string. It stops when the specific task is 
+completed. Therefore no validation occurs outside of this scope.  For example,
+```
+get_tld('......com') -> 'com'
+``` 
+as leading dots are not processed.
+split_domain method is based on get_sld method - it returns everything in 
+front of get_sld() as a prefix.
+Specifically to example above 
```

Comment:
@vadym-t you might also add here a non-edge case example. the split_domain() method offers a new capability to the library-- one that folks might get from other libraries -- but your only example is the edge. suggest something like:
split_domain allows you to recover the host, or prefix, of an SLD, for use in aggregation or analysis based on the labels. e.g., split_domain('www.googl.com')